src/editor/object_option.hpp

```diff
@@ -68,12 +68,29 @@ class ObjectOption
   const std::string m_key;
   const unsigned int m_flags;
 
+private:
+  BaseObjectOption(const BaseObjectOption&) = delete;
+  BaseObjectOption& operator=(const BaseObjectOption&) = delete;
+};
+
+template <typename T>
```

Comment:
[You can use default template arguments.](https://stackoverflow.com/questions/15373823/how-to-avoid-specifying-arguments-for-a-class-template-with-default-template-arg)
You might need to change in other places, not sure